src/state_machine/transitions.hpp

```diff
@@ -0,0 +1,118 @@
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#ifndef STATE_MACHINE_TRANSITIONS_HPP_
+#define STATE_MACHINE_TRANSITIONS_HPP_
+
+#include "data/data.hpp"
+#include "state_machine/state.hpp"
+#include "utils/logger.hpp"
+
+namespace hyped {
+
+namespace state_machine {
+
+using hyped::data::Batteries;
+using hyped::data::EmergencyBrakes;
+using hyped::data::Motors;
+using hyped::data::Navigation;
+using hyped::data::Sensors;
+using hyped::data::Telemetry;
+using utils::Logger;
+
+class State;  // Forward declaration
+
+//--------------------------------------------------------------------------------------
+// Emergency
+//--------------------------------------------------------------------------------------
+
+/*
+ * @brief   Returns the FailureStopped if there's an emergency and NULL otherwise.
+ */
+State *checkEmergencyStationary(Logger &log, EmergencyBrakes embrakes_data, Navigation nav_data,
+                                Batteries batteries_data, Telemetry telemetry_data,
+                                Sensors sensors_data, Motors motors_data);
+
+/*
+ * @brief   Returns the FailureBraking if there's an emergency and NULL otherwise.
+ */
+State *checkEmergencyMoving(Logger &log, EmergencyBrakes embrakes_data, Navigation nav_data,
+                            Batteries batteries_data, Telemetry telemetry_data,
+                            Sensors sensors_data, Motors motors_data);
+
+//--------------------------------------------------------------------------------------
+// Module Status
+//--------------------------------------------------------------------------------------
+
+/*
+ * @brief   Returns the Calibrating state if the command has been received and all modules are
+ *          initialised.
+ */
+State *checkModulesInitialised(Logger &log, EmergencyBrakes embrakes_data, Navigation nav_data,
+                               Batteries batteries_data, Telemetry telemetry_data,
+                               Sensors sensors_data, Motors motors_data);
+
+/*
+ * @brief    Returns the Ready state if all Embrakes, Navigation and Motors are ready.
+ */
+State *checkModulesReady(Logger &log, EmergencyBrakes embrakes_data, Navigation nav_data,
+                         Motors motors_data);
+
+//--------------------------------------------------------------------------------------
+// Telemetry Commands
+//--------------------------------------------------------------------------------------
+
+/*
+ * @brief   Returns the Accelerating state if the launch command has been received.
+ */
+State *checkLaunchCommand(Logger &log, Telemetry telemetry_data);
+
+/*
+ * @brief    Returns the Stopped state if the shutdown command has been received.
+ */
+State *checkShutdownCommand(Logger &log, Telemetry telemetry_data);
```

Comment:
I tried to separate the high-level behaviour and the details. As you've pointed out above, changing the return type to boolean might be a good idea. Yet, since those functions do more than just checking the command (i.e. logging), I think it's a good idea to have them grouped together. It should remove some noise in `state.cpp`.